snapcraft/internal/lxd.py

```diff
@@ -106,12 +111,19 @@ def _ensure_started(self):
             print('Stopping {}'.format(self._container_name))
             check_call(['lxc', 'stop', '-f', self._container_name])
 
+    def _install_packages(self, packages):
+        try:
+            self._wait_for_network()
+            self._container_run(['apt-get', 'update'])
+            self._container_run(['apt-get', 'install', '-y', *packages])
+        finally:
+            # Always remove apt lock in case we stop during install
+            self._container_run(['rm', '-f', '/var/lib/apt/lists/lock'])
```

Comment:
It was clear for me. I think @kyrofa needs to ctrl+c in the middle of an update, to see how things go crazy and non-user-friendly.